src/autoscaler/routes/routes.go

```diff
@@ -127,9 +135,15 @@ func newRouters() *AutoScalerRoute {
 	instance.publicApiProtectedRoutes.Path(PublicApiScalingHistoryPath).Methods(http.MethodGet).Name(PublicApiScalingHistoryRouteName)
 	instance.publicApiProtectedRoutes.Path(PublicApiMetricsHistoryPath).Methods(http.MethodGet).Name(PublicApiMetricsHistoryRouteName)
 	instance.publicApiProtectedRoutes.Path(PublicApiAggregatedMetricsHistoryPath).Methods(http.MethodGet).Name(PublicApiAggregatedMetricsHistoryRouteName)
-	instance.publicApiProtectedRoutes.Path(PublicApiPolicyPath).Methods(http.MethodGet).Name(PublicApiGetPolicyRouteName)
-	instance.publicApiProtectedRoutes.Path(PublicApiPolicyPath).Methods(http.MethodPut).Name(PublicApiAttachPolicyRouteName)
-	instance.publicApiProtectedRoutes.Path(PublicApiPolicyPath).Methods(http.MethodDelete).Name(PublicApiDetachPolicyRouteName)
+
+	instance.publicApiPolicyRouters = instance.publicApiRoutes.Path(PublicApiPolicyPath).Subrouter()
+	instance.publicApiPolicyRouters.Path("").Methods(http.MethodGet).Name(PublicApiGetPolicyRouteName)
+	instance.publicApiPolicyRouters.Path("").Methods(http.MethodPut).Name(PublicApiAttachPolicyRouteName)
+	instance.publicApiPolicyRouters.Path("").Methods(http.MethodDelete).Name(PublicApiDetachPolicyRouteName)
+
+	instance.publicApiCustomMetricsCredentialRouters = instance.publicApiRoutes.Path(PublicApiCustomMetricsCredentialPath).Subrouter()
+	instance.publicApiCustomMetricsCredentialRouters.Path("").Methods(http.MethodPut).Name(PublicApiCreateCustomMetricsCredentialRouteName)
+	instance.publicApiCustomMetricsCredentialRouters.Path("").Methods(http.MethodDelete).Name(PublicApiDeleteCustomMetricsCredentialRouteName)
```

Comment:
publicApiPolicyRouters and publicApiCustomMetricsCredentialRouters are protected by OAuth as well, so the current naming "publicApiProtectedRoutes" for scaling history and scaling metrics are not accurate enough.
Suggest to change:
* `publicApiRoutes`  -> `ApiOpenRoutes`
* `publicApiProtectedRoutes` -> `ApiRoutes`
* `publicApiPolicyRouters` -> `ApiPolicyRoutes`
* `publicApiCustomMetricsCredentialRouters` -> `ApiCredentialRoutes`